### PR TITLE
TASK-59276: Improve protection to protect REST services

### PR DESCRIPTION
**Pull request by hchaabene**
Prior this change, 2 rest API endpoints **exportNote()** and **cancelExportNote()** have @POST as HTTP method but doesn't write in DATABASE so they need to be changed to @GET

### CODE REVIEW

File: notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java [hunk 1]
```diff
@@ -114,11 +114,11 @@ public NotesRestService(NoteService noteService,
     cc.setNoStore(true);
   }
 
-  @GET
+  @PUT
```
Comment (rdenarie) [inline notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java]: Why did you change this one ? By reading the function, it seems to be a simple read (getNote)

File: notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java [hunk 2]
```diff
@@ -307,11 +307,11 @@ public Response getLatestDraftOfPage(@Parameter(description = "Note id", require
     }
   }
 
-  @GET
+  @POST
```
Comment (rdenarie) [inline notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java]: This one seems also a READ action

**Review by rdenarie — CHANGES_REQUESTED** [review]

**Review by azayati — CHANGES_REQUESTED** [review]
@hchaabene I don't think that we should change annotations, this is not what is asked by the US

File: notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java [hunk 3]
```diff
@@ -118,7 +118,7 @@ public NotesRestService(NoteService noteService,
   @Path("/note/{noteBookType}/{noteBookOwner:.+}/{noteId}")
   @Produces(MediaType.APPLICATION_JSON)
   @RolesAllowed("users")
-  @Operation(summary = "Get note by notes params", method = "GET", description = "This get the not if the authenticated user has permissions to view the objects linked to this note.")
+  @Operation(summary = "Get note by notes params", method = "PUT", description = "This get the not if the authenticated user has permissions to view the objects linked to this note.")
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java]: revert this change

File: notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java [hunk 4]
```diff
@@ -311,7 +311,7 @@ public Response getLatestDraftOfPage(@Parameter(description = "Note id", require
   @Path("/versions/{noteId}")
   @Produces(MediaType.APPLICATION_JSON)
   @RolesAllowed("users")
-  @Operation(summary = "Get versions of note by id", method = "GET", description = "This get the versions of a note if the authenticated user has permissions to view the objects linked to this note.")
+  @Operation(summary = "Get versions of note by id", method = "POST", description = "This get the versions of a note if the authenticated user has permissions to view the objects linked to this note.")
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java]: revert this change

**Review by azayati — CHANGES_REQUESTED** [review]
see review and change description according to the last review changes @hchaabene

**Review by azayati — APPROVED** [review]
LGTM

**Review by rdenarie — APPROVED** [review]
